lib/apiservers/engine/backends/container.go

```diff
@@ -1543,6 +1544,64 @@ func ContainerSignal(containerID string, sig uint64) error {
 	return nil
 }
 
+func getClientIPAddrs() []netlink.Addr {
+	l, err := netlink.LinkByName(clientIfaceName)
+	if err != nil {
+		log.Errorf("Could not look up link from client interface name %s due to error %s",
+			clientIfaceName, err.Error())
+		return nil
```

Comment:
why we are returning nil but reporting an error? If returning nil means err then I strongly suggest returning a pair ([]netlink.Addr, error) to make the intention of the function clear